metadata-models/src/main/pegasus/com/linkedin/ml/metadata/MLPrimaryKeyProperties.pdl

```diff
@@ -0,0 +1,40 @@
+namespace com.linkedin.ml.metadata
+
+import com.linkedin.common.Urn
+import com.linkedin.common.MLFeatureDataType
+import com.linkedin.common.VersionTag
+
+/**
+ * Properties associated with a MLPrimaryKey
+ */
+@Aspect = {
+  "name": "mlPrimaryKeyProperties"
+}
+record MLPrimaryKeyProperties {
+
+  /**
+   * Documentation of the MLPrimaryKey
+   */
+  description: optional string
+
+  /**
+   * Data Type of the MLPrimaryKey
+   */
+  dataType: optional MLFeatureDataType
```

Comment:
Should we require a data type?

Reply:
Yes – at least for compatibility with Feast, but also because feature primary keys are often integers or strings

Reply:
I mean, should this field be optional or required?

Reply:
I think required is fine – if a future data source is missing a datatype we can still annotate it as UNKNOWN